liquidctl/driver/asetek_pro.py

```diff
@@ -0,0 +1,256 @@
+        self._end_transaction_and_read(3)
+
+    def _write_color(self, colors):
+        """Write a list of colors to cycle between, the cycle pattern comes from the
+        value used for cycle speed"""
+        colorCount = 0
+        setColors = []
+        for color in colors:
+            colorCount = colorCount + 1
+            setColors = setColors + color
+        self._write([_PRO_CMD_WRITE_COLOR_LIST, colorCount] + setColors)
+        self._end_transaction_and_read(3)
+
+    def _get_fan_speeds(self):
+        """Read the RPM speed of the fans. Try to get the speeds for 3 fans as that is
+        how many the H150i PRO supports"""
+        speeds = []
+        for i in [0, 1, 2]:
+            self._write([_PRO_CMD_READ_FAN_SPEED, i])
+            msg = self._end_transaction_and_read(6)
+            if msg[0] != 0x41 or msg[1] != 0x12 or msg[2] != 0x34 or msg[3] != i:
+                _LOGGER.debug('Unable to get speed for fan id %d', i)
+                continue
+            speeds.append((i, (msg[4] << 8) + msg[5]))
+        return speeds
+
+    def _get_fan_indexes(self, channel):
+        if len(channel) > 3:
+            return [int(channel[3:]) - 1]
+        return [reading[0] for reading in self._get_fan_speeds()]
+
+    def initialize(self, pump_mode=None, **kwargs):
+        """Initialize the device."""
+        _LOGGER.debug('Pro configure device...')
+        if pump_mode == None:
+            return
+        else:
+            _LOGGER.debug('pump_mode %s', pump_mode)
+            self._write([_PRO_CMD_WRITE_PUMP_MODE, _PRO_PUMP_MODES.index(pump_mode)])
+            self._end_transaction_and_read(5)
+
+    def get_status(self, **kwargs):
+        """Get a status report.
+        Returns a list of `(property, value, unit)` tuples.
+        """
+        status = []
+        ### Can some Pro devices have more than one temp sensor?   OCL seems to code for that?
+        self._write([_PRO_CMD_READ_AIO_TEMP])
+        msg = self._end_transaction_and_read(6)
+        aio_temp = msg[3] + msg[4]/10
+        status.append(('Liquid temperature', aio_temp, '°C'))
+        speeds = self._get_fan_speeds()
+        for reading in speeds:
+            status.append(('Fan {} speed'.format(reading[0]), reading[1] | 0, 'rpm'))
+        self._write([_PRO_CMD_READ_PUMP_MODE])
+        msg = self._end_transaction_and_read(4)
+        pump_mode = msg[3]
+        status.append(('Pump mode', _PRO_PUMP_MODES[pump_mode], ""))
+        self._write([_PRO_CMD_READ_PUMP_SPEED])
+        msg = self._end_transaction_and_read(5)
+        pump_speed = (msg[3] << 8) + msg[4]
+        status.append(('Pump speed', pump_speed | 0, 'rpm'))
+        self._write([_PRO_CMD_READ_FIRMWARE])
+        msg = self._end_transaction_and_read(7)
+        firmware = '{}.{}.{}.{}'.format(*tuple(msg[3:7]))
+        status.append(('Firmware version', firmware, ''))
+        return status
+
+    def set_color(self, channel, mode, colors, time_per_color=1, time_off=None,
```

Comment:
I'm going to skip this method for now.

Reply:
Hm, I noticed this before: it would be better to take the `speed` parameter as a string (`'slow', 'normal', 'fast'`).

Reply:
I still find it suspicious that it only accepts the 3 speeds, @andytank123 did you try other speed values (related to one of my earlier comments). I know for one of my devices (I dont remember which one, it may been the `H100i Platinum` icue listed fast normal and slow, but I could actually send any value.

Reply:
It would make sense if the parameters for each speed value are timings for parts of the animations.  Perhaps in tens of seconds?